### ISSUE TITLE
Make horizontal-layout and vertical-layout overflow:hidden by default.

### ISSUE THREAD

**Issue by rolfsmeds** [vaadin/vaadin-ordered-layout#75]
### Description
`overflow:hidden` is sometimes required to make flexbox behave as expected. Since `vaadin-horizontal-layout` and `vaadin-vertical-layout` don't have this by default, they can sometimes behave weirdly, allowing their contents to grow larger than layout, which is hardly what you want in most cases. In Firefox this generally manifests as an unwated scroll area, while in Chrome the issue is sometimes invisible but affects mousewheel/touch scrolling.

Adding `overflow:hidden` to the `:host` would avoid these issues.

### Browsers Affected

- [x] Chrome
- [x] Firefox

**Comment by tomivirkki** [comment]
This would cut off any content that's meant to visually overflow the layout:

![Screenshot 2019-11-25 15 11 05](https://user-images.githubusercontent.com/1222264/69543310-f8fb8b80-0f95-11ea-9acf-85dd424fd8f0.png)

...would appear as follows:
![Screenshot 2019-11-25 15 11 26](https://user-images.githubusercontent.com/1222264/69543322-01ec5d00-0f96-11ea-85fb-223bde0525cd.png)

**Comment by rolfsmeds** [comment]
Hmm. That's a good point, didn't consider shadows etc. Perhaps there is some other trick we can do to avoid these issues. I'll try to find time to investigate at some point and update if I come up with something.

**Comment by lkraav** [comment]
> overflow:hidden is sometimes required to make flexbox behave as expected.

@rolfsmeds are you able to provision an example of such a case?

**Comment by rolfsmeds** [comment]
> @rolfsmeds are you able to provision a code example of such a case?

Yup, currently experiencing the issue in a project, but will try to find some time soon to isolate it to a simple example.
